utils/utils.py

```diff
@@ -173,42 +232,108 @@ def dataset_loading_and_splitting(
                 dataset=dataset_combined,
                 batch_size=config["batch_size"],
                 perc_train=config["perc_train"],
+                distributed_data_parallelism=distributed_data_parallelism,
             )
 
 
-def split_dataset(dataset: [], batch_size: int, perc_train: float):
+def split_dataset(
+    dataset: [],
+    batch_size: int,
+    perc_train: float,
+    distributed_data_parallelism: bool = False,
+):
     perc_val = (1 - perc_train) / 2
     data_size = len(dataset)
-    train_loader = DataLoader(
-        dataset[: int(data_size * perc_train)], batch_size=batch_size, shuffle=True
-    )
-    val_loader = DataLoader(
-        dataset[int(data_size * perc_train) : int(data_size * (perc_train + perc_val))],
-        batch_size=batch_size,
-        shuffle=True,
-    )
-    test_loader = DataLoader(
-        dataset[int(data_size * (perc_train + perc_val)) :],
-        batch_size=batch_size,
-        shuffle=True,
-    )
+
+    trainset = dataset[: int(data_size * perc_train)]
+    valset = dataset[
+        int(data_size * perc_train) : int(data_size * (perc_train + perc_val))
+    ]
+    testset = dataset[int(data_size * (perc_train + perc_val)) :]
+
+    if distributed_data_parallelism:
+
+        train_sampler = torch.utils.data.distributed.DistributedSampler(trainset)
```

Comment:
So you can still wrap the existing data with `DistributedSampler` without inheriting?